lib/cli/driver.js

```diff
@@ -262,12 +273,130 @@ class DriverCommand {
+    // if we're specifically requesting an update for a driver, make sure it's installed
+    if (!shouldUpdateAll && !this.config.isInstalled(driver)) {
+      throw new Error(`Driver '${driver}' was not installed, so can't be updated`);
+    }
+    const driversToUpdate = shouldUpdateAll ? Object.keys(this.config.installedDrivers) : [driver];
+
+    // 'errors' will have driver names as keys and error objects as values
+    const errors = {};
+
+    // 'updates' will have driver names as keys and update objects as values, where an update
+    // object is of the form {from: versionString, to: versionString}
+    const updates = {};
+
+    for (const d of driversToUpdate) {
+      try {
+        await spinWith(this.isJsonOutput, `Checking if driver '${d}' is updatable`, () => {
+          if (this.config.installedDrivers[d].installType !== INSTALL_TYPE_NPM) {
+            throw new Error(`Driver '${d}' ${NOT_NPM_MSG}`);
+          }
+        });
+        const update = await spinWith(this.isJsonOutput, `Checking if driver '${d}' needs an update`, async () => {
+          const update = await this.checkForDriverUpdate(d);
+          if (!(update.safeUpdate || update.unsafeUpdate)) {
+            throw new Error(`Driver '${d}' ${NO_UPDATES_MSG}`);
+          }
+          return update;
+        });
+        if (!unsafe && !update.safeUpdate) {
+          throw new Error(`Driver '${d}' has a major revision update ` +
+                          `(${update.current} => ${update.unsafeUpdate}), which could include ` +
+                          `breaking changes. If you want to apply this update, re-run with --unsafe`);
+        }
+        const updateVer = unsafe && update.unsafeUpdate ? update.unsafeUpdate : update.safeUpdate;
+        await spinWith(
+          this.isJsonOutput,
+          `Updating driver '${d}' from ${update.current} to ${updateVer}`,
+          async () => await this.updateDriver(d, updateVer)
+        );
+        updates[d] = {from: update.current, to: updateVer};
+      } catch (err) {
+        errors[d] = err;
+      }
+    }
+
+    log(this.isJsonOutput, 'Update report:');
+    for (const [d, update] of _.toPairs(updates)) {
+      log(this.isJsonOutput, `- Driver ${d} updated: ${update.from} => ${update.to}`.green);
+    }
+    for (const [d, err] of _.toPairs(errors)) {
+      if (new RegExp(NO_UPDATES_MSG).test(err) || new RegExp(NOT_NPM_MSG).test(err)) {
+        // if the error we received is not an unexpected one, just print it out in yellow
+        log(this.isJsonOutput, `- ${err.message}`.yellow);
+      } else {
+        // otherwise, make it pop with red!
+        log(this.isJsonOutput, `- Driver ${d} failed to update: ${err}`.red);
+      }
+    }
+
+    return {updates, errors};
+  }
+
+  /**
+   * Given a driver name, figure out what its highest possible version upgrade is, and also the
+   * highest possible safe upgrade.
+   *
+   * @param [string] driver - name of driver
```

Comment:
Please use curly brackets for types to align with the common documentation style